src/preprocess.rkt

```diff
@@ -72,7 +87,8 @@
        simplified
        (list initial))
    ;; Absolute value should happen before sorting
-   (append abs-instructions sort-instructions)))
+   ;; TODO: Does this order work?
+   (append abs-instructions negabs-instructions sort-instructions)))
```

Comment:
Yeah, I think you shouldn't be able to have both abs and nabs on the same variable (except in the weird special case where f(x) = 0? But then I don't think we'd keep the preprocessing step?)